app/components/facility/AddFacilityComponent.tsx

```diff
@@ -0,0 +1,134 @@
+import React, {useState} from 'react';
+import {Button, Modal} from 'react-bootstrap';
+import {JSONSchema6} from 'json-schema';
+import JsonSchemaForm, {IChangeEvent} from 'react-jsonschema-form';
+import moment from 'moment-timezone';
+import FormFieldTemplate from '../../containers/Forms/FormFieldTemplate';
+import FormObjectFieldTemplate from '../../containers/Forms/FormObjectFieldTemplate';
+
+interface Props {
+  handleAddFacility: any;
+  organisationRowId: string;
```

Comment:
Shouldn't this be a number?

Reply:
yep

Reply:
Hm. On second look, our facility table stores the reportingYear as a string. It's also stored as a varchar in swrs and used as a varchar in the init functions

Reply:
uh? This is 'organisationRowId', not reporting year. Also reportingYear should be a number everywhere, including in SWRS, where it was changed to a varchar only to accomodate a metabase bug

Reply:
oh yeah! Well I changed the organisationRowId to a number, but we do have a type problem with reportingYear, it is a varchar in most places, and an int in others

Reply:
Note added for checking into reportingYear type mismatch: https://youtrack.button.is/agiles/101-1/102-295?issue=GGIRCS-1111